### PR TITLE
Implement plugin-creator command.

### PR DESCRIPTION
**Pull request by matootie**
A command to initialize a new plugin directory with basic files to get started.

### CODE REVIEW

File: kiki/base/commands.py [hunk 1]
```diff
@@ -53,7 +61,74 @@ def run(**flags):
...
+            tasks_file.write('''"""
+Write any functions used by the plugin here.
+"""
+            ''')
+    # If we recieve an error due to an existing file...
+    except FileExistsError:
+        # Let the user know another plugin is using this name.
+        print("A plugin using this name already exists!")
+
+
```
[the remark this answers is not in the document]
Comment (matootie) [reply kiki/base/commands.py]: The small preview unfortunately doesn't show the entire function..

File: kiki/base/commands.py [hunk 2]
```diff
@@ -23,10 +25,16 @@ def run(**flags):
...
     # Grab our potential flags.
+    help = find_flag(flags, short="h", long="help")
     token = find_flag(flags, short="t", long="token")
     plugins = find_flag(flags, short="p", long="plugins")
+    # If the help tag was provided...
+    if help:
+        # Print help text.
+        print(run.__doc__)
+        # Exit.
+        return
```
[the remark this answers is not in the document]
Comment (matootie) [reply kiki/base/commands.py]: The small preview here doesn't show the all the code I wanted to highlight...

File: kiki/base/commands.py [hunk 3]
```diff
@@ -53,7 +61,74 @@ def run(**flags):
...
+        # Let the user know where they went wrong.
+        print("No name was provided.")
+        # Exit.
+        return
+    # Attempt to create a new plugin module.
+    try:
+        plugin_directory = "plugins/{}/".format(name)
+        os.makedirs("{}/".format(plugin_directory))
+        with open("{}/__init__.py".format(plugin_directory), "w") as init_file:
+            init_file.write("")
```
Comment (sventi555) [inline kiki/base/commands.py]: Is there a more elegant way to `touch` a file?

Comment (matootie) [reply kiki/base/commands.py]: I've seen a couple of different approaches. The simplest one involved a library called `pathlib`. I figured seeing as this approach is efficient for the other two files that are being written, it wouldn't make sense to import a completely different library for one of three files.

I suppose we could simplify the same approach like so...

*replace*
```python
with open("".format(plugin_directory), "w") as init_file:
    init_file.write("")
```
*with*
```python
open("{}/__init__.py".format(plugin_directory), 'w').close()
```

Thoughts?

Comment (sventi555) [reply kiki/base/commands.py]: I like the second more because it's clearer. But the first fits with the style of the other two. Take your pick honestly.

**Comment by matootie** [comment]
I've changed the code. Take a look and accept the changes when you can @sventi555. Merging is blocked unless a review approval is given by anyone but me.

**Pull request merged**